File: data-agent-frontend/src/api/sqlTrace.ts
/*
 * Copyright (C) 2026 github.com/MaloneTalk
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 * limitations under the License.
 */

import request from './request';
import type { ApiResponse } from './request';
import type { PageResponse } from './types';

export interface SqlTraceResponse {
  id: number;
  sessionId: string;
  traceId: string | null;
  datasourceName: string | null;
  databaseName: string | null;
  tableNames: string[];
  sql: string;
  rowCount: number | null;
  truncated: boolean;
  durationMs: number | null;
  createTime: string;
  /** 列表查询不返回；详情查询返回完整结果 JSON（columns/rows/totalRows/truncated）。 */
  resultJson: string | null;
}

export interface SqlTracePageQuery {
  page?: number;
  pageSize?: number;
}

export function getSqlTraces(sessionId: string, query: SqlTracePageQuery = {}) {
  return request.get<ApiResponse<PageResponse<SqlTraceResponse>>>(
    `/agent/session/${sessionId}/sql-trace`,
    { params: query },
  );
}

export function getSqlTraceDetail(sessionId: string, id: number) {
  return request.get<ApiResponse<SqlTraceResponse>>(`/agent/session/${sessionId}/sql-trace/${id}`);
}

File: data-agent-backend/src/main/java/io/github/malonetalk/entity/SqlTrace.java
/*
 * Copyright (C) 2026 github.com/MaloneTalk
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 * limitations under the License.
 */
package io.github.malonetalk.entity;

import java.time.LocalDateTime;
import lombok.Data;

/** execute_sql 工具的成功执行记录（表名与结果 JSON 冗余存储，避免数据源改名/删除后丢失上下文）。 */
@Data
public class SqlTrace {

    private Long id;
    private String sessionId;
    private Integer userId;
    private String traceId;
    private Integer datasourceId;
    private String datasourceName;
    private String databaseName;

    /** 逗号分隔的表名。 */
    private String tableNames;

    private String sqlText;
    private String resultJson;
    private Integer rowCount;
    private Boolean truncated;
    private Long durationMs;
    private LocalDateTime createTime;
}
